docs/source/examples.rst

```diff
@@ -1,10 +1,295 @@
 .. _examples_page:
 
+########
 Examples
-========
+########
 
-ns-3 simulation
+
+.. _ns3_connection:
+
+*******************
+Connetion with ns-3
+*******************
+
+We will demonstrate the cooperation of Reinforced-lib with external WiFi simulating software based on the example of
+ML controlled rate adaptation manager. To simulate the WiFi environment, we will use the a popular, research oriented
+network simulator - ns-3. To learn more about the simulator, we encourage to visit the
+`official website <https://www.nsnam.org/>`_ or read the
+`ns-3 tutorial <https://www.nsnam.org/docs/release/3.36/tutorial/html/index.html>`_.
+
+
+Environment setup
+=================
+
+In order to perform experiments with pythonic Reinforced-lib and C++ written ns-3, you need to setup the environment in
+following order:
+
+  * Favourite C++ compiler (we assume that you already have one in your dev stack)
+  * ns-3 (connection tested on ns-3.36 version)
+  * ns3-ai
+
+
+Installing ns-3
 ---------------
 
-Gym environment integration
+There are a few ways to install ns-3, all described on the `ns-3 wiki <https://www.nsnam.org/wiki/Installation>`_,
+but we recomend to install the ns-3 by cloning their git dev repository:
+
+.. code-block:: bash
+
+    git clone https://gitlab.com/nsnam/ns-3-dev.git
+
+Then change directory to newly created ``ns-3-dev/`` and build the ns-3:
+
+.. code-block:: bash
+
+    ./ns3 configure --enable-examples
+    ./ns3 build
+
+Once you have built ns-3 (with examples enabled), it should be easy to run the sample programs with the following command,
+such as:
+
+.. code-block:: bash
+
+    ./ns3 run wifi-simple-adhoc
+
+If the installation process was successful, you should now have two new ``.pcap`` files in the ``ns-3-dev/`` directory, namely
+``wifi-simple-adhoc-0-0.pcap`` and ``wifi-simple-adhoc-1-0.pcap``
+
+
+Installing ns3-ai
+-----------------
+
+The ns3-ai module interconnects the ns-3 and Reinforced-lib (or other python-writen software) by transferring data through
+the shared memory pool. The memory can be accessed by both sides thus making the connection. Read more on ns3-ai on the
+`official repository <https://github.com/hust-diangroup/ns3-ai>`_. To install the module colone the ns-3 repository to the
+``ns-3-dev/contrib/`` directory:
+
+.. code-block:: bash
+
+    cd $YOUR_NS3_PATH/contrib/
+    git clone https://github.com/hust-diangroup/ns3-ai.git
+
+To have ns3-ai working, you need to rebuild the ns-3, but before doing so, we will transfer some neccessary files to
+enable experimenting with the provided rate adaptation manager.
```

Comment:
Zastanawiam się jeszcze czy to w ogóle zadziała tak out-of-the-box, bo też ns3-ai dalej jest niezaktualizowane i nie za dobrze wspiera CMake w nowej wersji. Waham się czy nie dodać poniżej linku do naszej wersji ns3-ai i powiedzieć, że gdyby coś nie działało, to można od nas pobrać (z mojego forka), ustawić branch na ml4wifi i wtedy powinno być okey.

Reply:
No właśnie wiem, ale nie wiedziałem jak to zgrabnie ująć, też głupio nie linkować od autora

Reply:
Nie no, do autora trzeba koniecznie podlinkować. Chodzi mi tylko o dodatkowy wpis, że jakby co, to my testowaliśmy na własnym forku i można spróbować. To wszystko co jest do tej pory napisane i podlinkowane, to ja bym zostawił.

Reply:
Okej, dodałem warning pod tym, że jest niekompatybilna wersja i że tutaj jest alternatywa dla ns3.36